Source/WebCore/Modules/WebGPU/GPUDevice.cpp

```diff
@@ -108,7 +116,22 @@ Ref<GPUQueue> GPUDevice::queue() const
 
 void GPUDevice::destroy()
 {
+    if (m_destroyed)
+        return;
+
+    m_destroyed = true;
+    m_backing->destroy();
+    auto ref = GPUDeviceLostInfo::create(PAL::WebGPU::DeviceLostInfo::create(PAL::WebGPU::DeviceLostReason::Destroyed, "destroyed"_s));
+    m_lostPromise->resolve(WTFMove(ref));
+    m_adapter->setInvalid();
+}
+
+void GPUDevice::makeInvalid()
+{
+    m_destroyed = true;
     m_backing->destroy();
+    auto ref = GPUDeviceLostInfo::create(PAL::WebGPU::DeviceLostInfo::create(PAL::WebGPU::DeviceLostReason::Unknown, ""_s));
+    m_lostPromise->resolve(WTFMove(ref));
```

Comment:
Is this right? 🤔

Reply:
I think so, it passes the test. Is there an alternative?